main/src/org/destinationsol/CommonDrawer.java

```diff
@@ -29,124 +29,139 @@
-    }
-    SolMath.free(pos);
-  }
-
-  public void drawLine(TextureRegion tex, float x, float y, float angle, float len, Color col, float width) {
-    draw(tex, len, width, 0, width/2, x, y, angle, col);
-  }
-
-  public void drawLine(TextureRegion tex, Vector2 p1, Vector2 p2, Color col, float width, boolean precise) {
-    Vector2 v = SolMath.getVec(p2);
-    v.sub(p1);
-    drawLine(tex, p1.x, p1.y, SolMath.angle(v, precise), v.len(), col, width);
-    SolMath.free(v);
-  }
-
-  public void dispose() {
-    mySpriteBatch.dispose();
-    myFont.dispose();
-  }
-
-  public SpriteBatch getBatch(Texture texture, TextureAtlas.AtlasRegion tex) {
-    myTextureChecker.onSprite(texture, tex);
-    return mySpriteBatch;
-  }
-
-  public void setAdditive(boolean additive) {
-    int dstFunc = additive ? GL20.GL_ONE : GL20.GL_ONE_MINUS_SRC_ALPHA;
-    mySpriteBatch.setBlendFunction(GL20.GL_SRC_ALPHA, dstFunc);
-  }
+    public final float w;
+    public final float h;
+    public final float r;
+
+    private final SpriteBatch mySpriteBatch;
+    private final BitmapFont myFont;
+    private final float myOrigFontHeight;
+    private final TextureChecker myTextureChecker;
+    private final GlyphLayout layout;
+
+    public CommonDrawer() {
+        myTextureChecker = new TextureChecker();
+        w = Gdx.graphics.getWidth();
+        h = Gdx.graphics.getHeight();
+        r = w / h;
+        mySpriteBatch = new SpriteBatch();
+
+        final FileHandle fontFile = FileManager.getInstance().getFontsDirectory().child("main.fnt");
+        myFont = new BitmapFont(fontFile, true);
+        myFont.setUseIntegerPositions(false);
+
+        myOrigFontHeight = myFont.getXHeight();
+
+        layout = new GlyphLayout();
+    }
+
+    public void setMtx(Matrix4 mtx) {
+        mySpriteBatch.setProjectionMatrix(mtx);
+    }
+
+    public void begin() {
+        mySpriteBatch.begin();
+    }
+
+    public void end() {
+        myTextureChecker.onEnd();
+        mySpriteBatch.end();
+    }
+
+    public void drawString(String s, float x, float y, float fontSize, Color col) {
+        if (s == null) return;
+        myTextureChecker.onString(myFont.getRegion().getTexture());
+        myFont.setColor(col);
+        myFont.getData().setScale(fontSize / myOrigFontHeight);
+        myFont.draw(mySpriteBatch, s, x, y);
+    }
+
+    public void drawStringCentered(String s, float x, float y, float fontSize, Color col) {
+        if (s == null) return;
+        myTextureChecker.onString(myFont.getRegion().getTexture());
+        myFont.setColor(col);
+        myFont.getData().setScale(fontSize / myOrigFontHeight);
+
+        // http://www.badlogicgames.com/wordpress/?p=3658
+        layout.reset();
+        layout.setText(myFont, s);
+        x -= layout.width / 2;
+        y -= layout.height / 2;
+        myFont.draw(mySpriteBatch, layout, x, y);
+    }
+
+    public void drawStringLeftAligned(String s, float x, float y, float fontSize, Color col) {
```

Comment:
Added a method to left align text. Just doesn't shift the text to the horizontal center of the layout.